### PR TITLE
[R4R] Add rb-jekyll-theme-* themes

### PR DESCRIPTION
**Pull request by sodre**
ref: #9315

Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/rb-jekyll-swiss/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+test:
+  commands:
+    - gem list -i {{ name }} -v {{ version }}
+
+about:
+  home: https://rubygems.org/gems/{{ name }}
+  license: MIT
+  license_family: MIT
+  license_file: {{ name }}-{{ version }}/LICENSE.txt
+  summary: A Jekyll plugin to render a project's README as the site's index.
```
Comment (bgruening) [inline recipes/rb-jekyll-swiss/meta.yaml]:
```suggestion
  summary: A bold typographic theme for Jekyll, inspired by Swiss design.
```

File: recipes/rb-jekyll-theme-cayman/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+test:
+  commands:
+    - gem list -i {{ name }} -v {{ version }}
+
+about:
+  home: https://rubygems.org/gems/{{ name }}
+  license: CC0-1.0
+  license_file: {{ name }}-{{ version }}/LICENSE
+  summary: Architect is a Jekyll theme for GitHub Pages
```
Comment (sodre) [inline recipes/rb-jekyll-theme-cayman/meta.yaml]:
```suggestion
  summary: Cayman is a Jekyll theme for GitHub Pages
```

File: recipes/rb-jekyll-theme-architect/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    - gem list -i {{ name }} -v {{ version }}
+
+about:
+  home: https://rubygems.org/gems/{{ name }}
+  license: CC0-1.0
+  license_file: {{ name }}-{{ version }}/LICENSE
+  summary: Architect is a Jekyll theme for GitHub Pages
+
+  doc_url: https://www.rubydoc.info/gems/{{ name }}
+  dev_url: https://github.com/pages-theme/{{ theme_name }}
```
Comment (sodre) [inline recipes/rb-jekyll-theme-architect/meta.yaml]: This needs to be done for all recipes
```suggestion
  dev_url: https://github.com/pages-themes/{{ theme_name }}
```

**Comment by sodre** [comment]
@conda-forge/staged-recipes the rb-jekyll-theme-* recipes are ready for review/merge

**Pull request merged**